### DISCUSSION TITLE
Intel Realsense problem

### DISCUSSION THREAD

**Discussion by anbarsanti** [ultralytics/ultralytics#17822] [Q&A] [answered]
Hi, I was using intel real sense camera D435I and integrated it with YOLOv11. 

After some time, my code no longer works. 

Here is the code:

```
# Initialize RealSense Pipeline
pipe = rs.pipeline()
cfg  = rs.config()

# Enable color stream (and depth if you want)
cfg.enable_stream(rs.stream.color, 640,480, rs.format.bgr8, 30)
# cfg.enable_stream(rs.stream.depth, 640,480, rs.format.z16, 30)

# Start Streaming
pipe.start(cfg)

while True:
    frame = pipe.wait_for_frames()
    color_frame = frame.get_color_frame()

    # Convert images to numpy arrays
    color_image = np.asanyarray(color_frame.get_data())

    # Run YOLOv8 OBB inference on the frame
    results = model.track(frame, stream=True, show=True, persist=True, tracker='bytetrack.yaml')  # Tracking with byteTrack

    # Process and visualize the results of Object Tracking with YOLO
    for r in results:
        annotated_frame = r.plot()

        # Display the annotated frame
        cv2.imshow("YOLOv11 OBB Inference - Realsense Camera", annotated_frame)

    # Break the loop if 'q' is pressed
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

# Stop Streaming
pipe.stop()
cv2.destroyAllWindows()
```

And here is the error message:
  File "/home/anbarsanti/Dropbox/YOLOv11/track.py", line 185, in <module>
    **for r in results:**
  File "/home/anbarsanti/anaconda3/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 36, in generator_context
    response = gen.send(None)
               ^^^^^^^^^^^^^^
  File "/home/anbarsanti/anaconda3/lib/python3.11/site-packages/ultralytics/engine/predictor.py", line 227, in stream_inference
    self.setup_source(source if source is not None else self.args.source)
  File "/home/anbarsanti/anaconda3/lib/python3.11/site-packages/ultralytics/engine/predictor.py", line 199, in setup_source
    self.dataset = load_inference_source(
                   ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/anbarsanti/anaconda3/lib/python3.11/site-packages/ultralytics/data/build.py", line 187, in load_inference_source
    source, stream, screenshot, from_img, in_memory, tensor = check_source(source)
                                                              ^^^^^^^^^^^^^^^^^^^^
  File "/home/anbarsanti/anaconda3/lib/python3.11/site-packages/ultralytics/data/build.py", line 169, in check_source
    raise TypeError("Unsupported image type. For supported types see https://docs.ultralytics.com/modes/predict")
TypeError: Unsupported image type. For supported types see https://docs.ultralytics.com/modes/predict

Could you help?

Thank you

**Comment by UltralyticsAssistant** [comment]
👋 Hello @anbarsanti, thank you for sharing your issue and for your interest in using Ultralytics 🚀! Below are some pointers and resources that might assist you while an Ultralytics engineer takes a closer look at your report.

From your description, it seems like you're encountering a 🐛 **Bug**. When reporting an issue, kindly ensure you're providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) (MRE). Your shared code snippet is helpful; however, double-check if additional context (such as your `YOLOv11` model setup or the input structure) or modifications you’ve made to the pipeline could be relevant as well.

## Recommendations

1. ### Upgrade
   Ensure you are using the latest version of the `ultralytics` package. This helps verify that the issue isn't already resolved in the latest release. You can upgrade using the following command:
   ```bash
   pip install -U ultralytics
   ```

2. ### Documentation
   The error traceback suggests an issue with the **image type** being passed as the source. Please review the documentation on [supported input sources](https://docs.ultralytics.com/modes/predict/) to confirm your setup is aligned with the expected formats.

3. ### Environments
   YOLO models, including enhanced versions such as `YOLOv11`, run seamlessly in various environments, ensuring compatibility across platforms. If your environment might be a factor, consider testing it in one of the following verified environments where all dependencies are preconfigured:
   - **Notebooks** with free GPU:  
     <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a>  
     <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>  
     <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
   - **Cloud VM**: [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) | [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)  
   - **Docker Image**: Use [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>  

4. ### Debugging the Data Pipeline
   - Review the `frame` object that's being passed to the `model.track()` function. Ensure that it's in a valid image or video format compatible with YOLO.
   - Convert your `color_image` to an acceptable format before inference if necessary, e.g., saving or structuring the numpy array as an image or frame that YOLO recognizes.

## Community Support
📢 In the meantime, feel free to engage with the Ultralytics community for real-time support or discussions:
- Join us on [Discord](https://ultralytics.com/discord) 🎧 for chat-based assistance.
- Participate in discussions on our [Discourse](https://community.ultralytics.com) platform.
- Share your insights or questions on our [Subreddit](https://reddit.com/r/ultralytics) to connect with the broader community.

## Status and CI Verification
You can also confirm that all tests are currently passing by checking the [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) badge below:
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  

An Ultralytics engineer will assist you soon to help resolve your issue. Thank you for your patience and for being part of the Ultralytics community! 🚀✨

**Comment by glenn-jocher** [comment, marked as answer]
@anbarsanti the error indicates that the `frame` object passed to `model.track()` is not in a format supported by YOLOv11. Instead of passing `frame` directly, convert `color_image` (which is the actual numpy array) for inference. Update the `model.track()` call to use `color_image`. Let me know if you need further clarification! For supported formats, refer to [YOLO Predict Modes](https://docs.ultralytics.com/modes/predict/).

**Reply by anbarsanti** [reply]
thank you so much @glenn-jocher !

**Reply by glenn-jocher** [reply]
@anbarsanti you're welcome! Credit goes to the YOLO community and the Ultralytics team for their hard work. Let us know if you have any further questions!
